src/quo/components/wallet/token_input/view.cljs

```diff
@@ -92,36 +84,49 @@
                                         (oops/oget "nativeEvent.selection")
                                         (js->clj :keywordize-keys true)
                                         (on-selection-change))))]
-    (fn [{:keys [token customization-color show-keyboard? crypto? currency value error? selection]
+    (fn [{:keys [token customization-color show-keyboard? crypto? currency value error? selection
+                 handle-on-swap]
           :or   {show-keyboard? true}}]
-      (let [theme (quo.theme/use-theme)]
-        [rn/view {:style {:flex 1}}
-         [rn/pressable {:on-press on-token-press}
-          [token/view
-           {:token token
-            :size  :size-32}]]
-         [rn/pressable
-          {:style    style/text-input-container
-           :on-press focus-input}
-          [rn/text-input
-           (cond-> {:style                    (style/text-input theme error?)
-                    :placeholder-text-color   (style/placeholder-text theme)
-                    :auto-focus               true
-                    :ref                      set-ref
-                    :placeholder              "0"
-                    :keyboard-type            :numeric
-                    :max-length               12
-                    :on-change-text           handle-on-change-text
-                    :selection-color          customization-color
-                    :show-soft-input-on-focus show-keyboard?
-                    :on-selection-change      handle-selection-change
-                    :selection                (clj->js selection)}
-             controlled-input?       (assoc :value value)
-             (not controlled-input?) (assoc :default-value value-internal))]]
-         [token-label
-          {:theme theme
-           :text  (if crypto? token currency)
-           :value (if controlled-input? value value-internal)}]]))))
+      (let [theme        (quo.theme/use-theme)
+            window       (rn/get-window)
+            window-width (:width window)]
+        [rn/pressable
+         {:style    {:width          "100%"
+                     :flex-direction :row}
+          :on-press on-token-press}
+         [token/view
+          {:token token
+           :size  :size-32}]
+         [rn/view {:style (style/input-container window-width)}
+          [rn/pressable
+           {:style    style/text-input-container
+            :on-press focus-input}
+           [rn/text-input
+            (cond-> {:style                    (style/text-input theme error?)
+                     :placeholder-text-color   (style/placeholder-text theme)
+                     :auto-focus               true
+                     :ref                      set-ref
+                     :placeholder              "0"
```

Comment:
is this placeholder correct? Probably should be an i81n label so it can be set.

Reply:
I think it is fine, the placeholder should be "0" in this case as this is the token amount input